libraries/AP_AHRS/AP_AHRS_NavEKF.cpp

```diff
@@ -1509,19 +1549,21 @@ bool AP_AHRS_NavEKF::resetHeightDatum(void)
     
     switch (ekf_type()) {
 
-    case 2:
-    default: {
+    case EKFType::NONE:
```

Comment:
This is a behavioural change; we return false here instead of whatever the EKF2 said.  This will be important for compiling without EKF2.